website/docs/feature-flag-tutorials/golang/implementing-feature-flags-golang.md

```diff
@@ -0,0 +1,290 @@
+
+## 2. Install a local feature flag provider
+
+In this section, we'll install Unleash, run the instance locally, log in, and create a feature flag. If you prefer, you can use other tools instead of Unleash, but you'll need to update the code accordingly. The basic steps will probably be the same.
+
+Use Git to clone the Unleash repository and Docker to build and run it. Open a terminal window and run the following commands:
+
+```
+git clone https://github.com/unleash/unleash.git
+cd unleash
+docker compose up -d
+```
+
+You will now have Unleash installed onto your machine and running in the background. You can access this instance in your web browser at [http://localhost:4242](http://localhost:4242).
+
+Log in to the platform using these credentials:
+
+```
+Username: admin
+Password: unleash4all
+```
+
+Click the 'New feature flag' button to create a new feature flag.
+
+![Create a new feature flag](../ruby/new-ff.png)
+
+Call it `graphql-api` and enable it in the `development` environment.
+
+![A feature flag called `graphql-api` is now visible.](../ruby/enable-ff.png)
+
+Everything's now setup on the Unleash side. Let's go to the code now.
+
+## 3. Grab country information from the REST Countries API
+
+We'll use the standard `net/http` package to make our HTTP requests and the Unleash SDK to connect to your local Unleash instance and retrieve your feature flag.
+
+Open a new tab in your terminal, and create a new folder (NOT in the unleash folder).
+
+```sh
+mkdir unleash-go
+cd unleash-go
+go mod init unleash-demo
+```
+
+Install dependencies:
+
+```sh
+go get github.com/Unleash/unleash-client-go/v4
+```
+
+Then, let's make sure our setup is working. Let's make a call to the REST API to retrieve information about a country using its country code. Create a file named `main.go`:
+
+```go
+package main
+
+import (
+    "encoding/json"
+    "fmt"
+    "io"
+    "log"
+    "net/http"
+)
+
+type Country struct {
+    Name    string `json:"name"`
+    Capital string `json:"capital"`
+}
+
+func main() {
+    resp, err := http.Get("https://restcountries.com/v2/alpha/no")
+    if err != nil {
+        log.Fatal(err)
+    }
+    defer resp.Body.Close()
+
+    body, _ := io.ReadAll(resp.Body)
+    var country Country
+    json.Unmarshal(body, &country)
+
+    fmt.Printf("Country: %s, Capital: %s\n", country.Name, country.Capital)
+}
+```
+
+Run the code:
+
+```sh
+go run main.go
+```
+
+You should see `Country: Norway, Capital: Oslo` in your terminal.
+
+## 4. Add the GraphQL endpoint
+
+The point of this tutorial is to mimic a real-world scenario where, based on a boolean feature flag, you would migrate from a REST API to a GraphQL one. So far, we've just used REST. Now, let's add the GraphQL version. The GraphQL endpoint is `https://countries.trevorblades.com/` (Thanks Trevor!).
```

Comment:
Should we introduce the GraphQL API we're using at the same time as we introduce the REST API?